### PR TITLE
RepGraph Test Actor when there is no PC

### PR DESCRIPTION
**Pull request by improbable-danny**
# Description
A RepGraph test actor which is always replicated to Spatial

Primary reviewers
@m-samiec

### CODE REVIEW

File: Game/Source/GDKTestGyms/Private/TestGymsReplicationGraph.cpp [hunk 1]
```diff
@@ -101,6 +103,7 @@ void UTestGymsReplicationGraph::InitGlobalActorClassSettings()
 	AddInfo(AReplicationGraphDebugActor::StaticClass(), EClassRepNodeMapping::NotRouted);				// Not supported
 	AddInfo(AInfo::StaticClass(), EClassRepNodeMapping::RelevantAllConnections);	// Non spatialized, relevant to all
 	AddInfo(ReplicatedBPClass, EClassRepNodeMapping::Spatialize_Dynamic);		// Add our replicated base class to ensure we don't miss out-of-memory bp classes
+	AddInfo(AReplicatedTestActorBase_RepGraph::StaticClass(), EClassRepNodeMapping::AlwaysReplicate);
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/Private/TestGymsReplicationGraph.cpp:106]: I think we only want to do this in spatial

File: Game/Source/GDKTestGyms/Public/TestGymsReplicationGraph.h [hunk 2]
```diff
@@ -61,6 +62,9 @@ class UTestGymsReplicationGraph : public USpatialReplicationGraph
 	UPROPERTY()
 	TSubclassOf<AActor> ReplicatedBPClass;
 
+	UPROPERTY()
+	TSubclassOf<AActor> ReplicatedTestActorBaseClass;
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/Public/TestGymsReplicationGraph.h:66]: Not used

File: Game/Source/GDKTestGyms/Public/TestGymsReplicationGraph.h [hunk 3]
```diff
@@ -20,6 +20,7 @@ enum class EClassRepNodeMapping : uint32
 {
 	NotRouted,						// Doesn't map to any node. Used for special case actors that handled by special case nodes (UTestGymsReplicationGraphNode_PlayerStateFrequencyLimiter)
 	RelevantAllConnections,			// Routes to an AlwaysRelevantNode or AlwaysRelevantStreamingLevelNode node
+	AlwaysReplicate,				// Always replicated to Spatial
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/Public/TestGymsReplicationGraph.h:23]: Can you flesh the comment out a bit about why this is different to RelevantAllConnections

File: Game/Source/GDKTestGyms/Private/TestGymsReplicationGraph.cpp [hunk 4]
```diff
@@ -106,6 +106,12 @@ void UTestGymsReplicationGraph::InitGlobalActorClassSettings()
 	{
 		// Game mode is replicated in spatial, ensure it is always replicated
 		AddInfo(AGameModeBase::StaticClass(), EClassRepNodeMapping::RelevantAllConnections);
+		// Add always replicated test actor
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/Private/TestGymsReplicationGraph.cpp:109]:
```suggestion
		// Add always replicated test actor. Use soft class path to work around module dependencies.
```

**Review by m-samiec — APPROVED** [review]

**Review by razvan-at-improbable — APPROVED** [review]
Good work, useful for multiple tests, not just this one, thanks! LGTM

**Pull request merged**